editoast/src/core/simulation.rs

```diff
@@ -66,6 +66,47 @@ pub struct PhysicsRollingStock {
     pub raise_pantograph_time: Option<u64>,
 }
 
+pub struct SimulationParameters {
+    pub tonnage: Option<f64>,
+    pub total_length: Option<f64>,
+}
+
+impl PhysicsRollingStock {
+    /// Create a new PhysicsRollingStock from a RollingStockModel and SimulationParameters
+    pub fn new(traction_engine: RollingStockModel, params: SimulationParameters) -> Self {
+        let traction_engine_length = traction_engine.length * 1000.0; // RollingStockModel length is in km, convert to m
```

Comment:
The [schema doc of RollingStock](https://github.com/OpenRailAssociation/osrd/blob/78160a257541d772fb4df95ff92b3218b463ea2d/python/osrd_schemas/osrd_schemas/rolling_stock.py#L205) says that the length is in meters.
Additionally the documentation of `PhysicsRollingStock` states the the length is in millimeters: https://github.com/OpenRailAssociation/osrd/blob/78160a257541d772fb4df95ff92b3218b463ea2d/editoast/src/core/simulation.rs#L41-L42

I guess this is why the `* 1000` works but the comments are misleading.

Reply:
yes, it looks like a typo in the comment

Reply:
Well if the simulation's length unit is actually meters as you corrected in the comment, I don't get the `* 1000` at all. Take `2TGVA` which `length` in the database is `476`. I hardly believe that's kilometers 😅

Reply:
The * 1000 was there before my changes as well. I guess I didn’t understand it correctly

Reply:
Maybe @flomonster can help to undestand this ?

Reply:
Is `params.total_length` in meters or in millimeters? If millimeters, it should probably be an int, and if meters, the code is wrong.

Reply:
I agree, we should start using types with units. Whenever we are using `mm` it should be stored as `u64`

Reply:
I looked into this library, and it's great for computations, but not very practical for API building and Serde integration is not easy. Maybe we should organize a workshop ?